kqueen/storages/etcd.py

```diff
@@ -592,24 +593,24 @@ def save(self, validate=True, assign_id=True):
             bool: `True` if model was saved without errors, `False` otherwise.
 
         """
+        with etcd.Lock(current_app.db.client, 'customer1'):
+            if assign_id:
+                self.verify_id()
 
-        if assign_id:
-            self.verify_id()
+            validation_status, validation_msg = self.validate()
```

Comment:
maybe wrap it in try-catch block?

Reply:
I just add with statement here, just the indent is changed. I agree, that we need to add try-catch, but first we need to raise an exception in validate method instead of returning Boolean value